### PR TITLE
Implement text marks

### PR DESCRIPTION
**Pull request by kadamwhite**
### Description of the problem this pull request fixes

Add support for Text marks and initialize the scene with a dummy text mark instance

Changes proposed in this pull request:
- Add Text mark with associated manipulators and properties panel. Supported properties should have parity with lyra 1.
- Permit properties window to scroll vertically when it doesn't fit on my teeny laptop screen (very rough impl)

*\* If submitting code for review: **
#### This PR includes:
- [x] Tests
- [x] functional comments
- [ ] high level description of any new components
### To-Do:
- [x] define connectors
- [x] define channels
- [x] tests!
- [ ] ~~Move hard-coded option dropdowns into vega schema~~

LYRA-226

For #227

### CODE REVIEW

**Comment by kadamwhite** [comment]
text smoke

**Comment by arvind** [comment]
One other direct-manipulation interaction that would be nice: being able to rotate the text when the command-key is pressed and you drag around a handle. This sort of interaction is present in Keynote, Omnigraffle, etc. for example.

**Comment by kadamwhite** [comment]
Per discussion with @arvind in Slack, hard-coding options for alignment, font etc in the property inspector components is sufficient for now. I will open an issue to document moving to json schema.

**Comment by kadamwhite** [comment]
@arvind what do you think about doing it through an additional handle? (not sure if it's possible to customize that cursor to be something other than a down-arrow though)
text rotation
I suggest this because I'm not aware of any other primitives we currently support having alternate handle-based manipulation modes given modifier keys, so I'm concerned about discoverability

File: src/js/components/inspectors/Text.jsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          field={update.text.field}
+          scale={update.text.scale}
+          signal={update.text.signal} />
+
+        <h3>Font</h3>
+
+        <Property name="font" label="Style"
+          primitive={primitive}
+          type="select"
+          opts={['Helvetica', 'Verdana', 'Georgia', 'Palatino', 'Garamond', 'Trebuchet MS']}
```
Comment (deathbearbrown) [inline src/js/components/inspectors/Text.jsx]: I don't know if this matters, but symbol mark contains all it's special info where the mark is defined. 
https://github.com/vega/lyra/blob/lyra2/src/js/model/primitives/marks/Symbol.js#L56

called in the inspector here: https://github.com/vega/lyra/blob/lyra2/src/js/components/inspectors/Symbol.jsx#L35

Comment (kadamwhite) [inline src/js/components/inspectors/Text.jsx]: That's not a bad pattern, I'll switch to that to keep the domain of these controls out of the component code. Thanks, @deathbearbrown !

Comment (kadamwhite) [inline src/js/components/inspectors/Text.jsx]: @arvind added an issue for the JSON schema q over in #284

File: src/scss/inspector.scss [hunk 2]
```diff
@@ -5,6 +5,7 @@
   bottom: 0;
   width: $inspector-width;
   background-color: $lblue-bg;
+  overflow-y: scroll;
```
Comment (deathbearbrown) [inline src/scss/inspector.scss:8]: +1 lol

**Comment by deathbearbrown** [comment]
so far what I see here looks good.

**Comment by deathbearbrown** [comment]
I know what that console warning is:
https://github.com/vega/lyra/blob/lyra2/src/js/model/rules/VLSingle.js#L10
You need to add `text: 'text'` here

**Comment by kadamwhite** [comment]
@arvind @deathbearbrown This one's in your court to review/merge; ping me in slack if any further changes are needed. Thanks for tracking down that bug, Sue!

**Comment by deathbearbrown** [comment]
+1 from me, I functionally tested and reviewed the code yesterday.  It just needed that array updated to work perfect, which you did :)

**Pull request merged**